src/GraphQL/Conversion/ListConverterFactories/ImmutableListConverters.cs

```diff
@@ -0,0 +1,23 @@
+#if NETCOREAPP1_0_OR_GREATER
+using System.Collections.Immutable;
+
+namespace GraphQL.Conversion;
+
+internal sealed class ImmutableArrayListConverterFactory : ListConverterFactoryBase
+{
+    public override Func<object?[], object> Create<T>()
+        => list => ImmutableArray.CreateRange(list.Cast<T>());
+}
+
+internal sealed class ImmutableListConverterFactory : ListConverterFactoryBase
+{
+    public override Func<object?[], object> Create<T>()
+        => list => ImmutableList.CreateRange(list.Cast<T>());
+}
+
+internal sealed class ImmutableSetListConverterFactory : ListConverterFactoryBase
+{
+    public override Func<object?[], object> Create<T>()
+        => list => ImmutableHashSet.CreateRange(list.Cast<T>());
+}
+#endif
```

Comment:
Sample immutable implementations.  (I'm removing from this PR for the moment; we can discuss further.)